https/tls_config.go

```diff
@@ -70,27 +81,60 @@ func ConfigToTLSConfig(c *TLSStruct) (*tls.Config, error) {
 	if c.TLSCertPath == "" {
 		return nil, errors.New("missing cert_file")
 	}
+
 	if c.TLSKeyPath == "" {
 		return nil, errors.New("missing key_file")
 	}
-	cfg := &tls.Config{
-		MinVersion: tls.VersionTLS12,
-	}
+
 	loadCert := func() (*tls.Certificate, error) {
 		cert, err := tls.LoadX509KeyPair(c.TLSCertPath, c.TLSKeyPath)
 		if err != nil {
 			return nil, errors.Wrap(err, "failed to load X509KeyPair")
 		}
 		return &cert, nil
 	}
+
 	// Confirm that certificate and key paths are valid.
 	if _, err := loadCert(); err != nil {
 		return nil, err
 	}
+
+	minVersion, err := pickMinTLSVersion(c.MinVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	maxVersion, err := pickMaxTLSVersion(c.MaxVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	cfg := &tls.Config{
+		MinVersion:               minVersion,
+		MaxVersion:               maxVersion,
+		PreferServerCipherSuites: c.PreferServerCipherSuites,
```

Comment:
This should always be `true`. No need for a config option.

* https://blog.bracebin.com/achieving-perfect-ssl-labs-score-with-go
* https://blog.cloudflare.com/exposing-go-on-the-internet/

Reply:
We don't know what vulnerabilities might come up in future that require flipping a flag at short notice, that's a major point of a strategy of having everything configurable. Even if current best practices are that this should always be true, it should still be configurable.

Reply:
I had a chat with someone about this setting.

There's no security risk here, but there is a compatibility risk. With "Server Chooses", the risk is that the client will fail to connect.

The recommendation I got was based on this:

> But if you control the client and server, then go high. This isn’t your homepage.

Since our primary client is Prometheus, we can keep this as "Server Chooses" on the exporter side.

Reply:
That's an argument about a default, not that it shouldn't be configurable.

Keep in mind also that we do not control the clients, as this will cover things like the Prometheus HTTP APIs and Pushgateway.

If you want to propose a policy where some settings are configurable and some are not then you need to explain how we decide that e.g. max version is configurable but this isn't.